.github/workflows/im-build-test-deploy.yaml

```diff
@@ -119,9 +119,21 @@ jobs:
       - name: Set hostnames based on environment
         run: |
           echo "API_HOSTNAME=api.im.$ENVIRONMENT.test.c.dhis2.org" >> $GITHUB_ENV
-          echo "API_URL=https://api.im.$ENVIRONMENT.test.c.dhis2.org" >> $GITHUB_ENV
           echo "UI_HOSTNAME=im.$ENVIRONMENT.test.c.dhis2.org" >> $GITHUB_ENV
 
+      - name: Set API_URL for im-web-client
+        if: github.repository == 'dhis2-sre/im-web-client'
+        run: |
+          # default to dev environment
```

Comment:
In what scenario is this happening? Would it make sense to adapt how we set the `ENVIRONMENT` var in the first place (in steps like `- name: "Environment: dev"`)? There might be other steps where defaulting to `dev` will make sense.

Reply:
We would be defaulting to the `dev` API environment whenever there's no API environment deployed from a matching branch. Let's say we have a PR on the web-client from branch `feat/something`, if there's no API environment with that name already up, we'll be using the `dev` one.

I think setting the `ENVIRONMENT` var is separate from these, unless I'm missing something.

> There might be other steps where defaulting to dev will make sense.

Can you share an example? 🤔

Reply:
I don't have an example. It was more of a question since it felt we are defaulting to `ENVIRONMENT=dev`. I was wondering whether it makes sense to default `ENVIRONMENT` to dev for all steps. This `ENVIRONMENT` would then be used by any other step.